azurelinuxagent/ga/update.py

```diff
@@ -766,6 +769,34 @@ def _send_heartbeat_telemetry(self, protocol):
             logger.info(u"[HEARTBEAT] Agent {0} is running as the goal state agent {1}", CURRENT_AGENT, debug_log_msg)
             self._last_telemetry_heartbeat = datetime.utcnow()
 
+    @staticmethod
+    def _ensure_extension_telemetry_state_configured_properly(protocol):
+        for name, path in list_agent_lib_directory(skip_agent_package=True):
+
+            try:
+                handler_instance = ExtHandlersHandler.get_ext_handler_instance_from_path_if_valid(name=name,
+                                                                                                  path=path,
+                                                                                                  protocol=protocol)
+                if handler_instance is not None:
+                    # Recreate the HandlerEnvironment for existing extensions on startup.
+                    # This is to ensure that existing extensions can start using the telemetry pipeline if they support
+                    # it and also ensures that the extensions are not sending out telemetry if the Agent has to disable the feature.
+                    handler_instance.create_handler_env()
+
+            except Exception:
+                # Ignore errors if any
+                continue
+
+        try:
+            if not is_extension_telemetry_pipeline_enabled():
+                # If extension telemetry pipeline is disabled, ensure we delete all existing extension events directory
+                # because the agent will not be listening on those events.
+                extension_event_dirs = glob.glob(os.path.join(conf.get_ext_log_dir(), "*", EVENTS_DIRECTORY))
```

Comment:
Why is `*` needed here, I thought the path for extension events is: `/var/log/azure/extension_name/events`?

Reply:
Ah, I think I got it. The `*` corresponds to any `extension_name`.

Reply:
Yeah exactly, its to capture every extension event dir and delete it